crates/goose/src/agents/extension_manager.rs

```diff
@@ -224,7 +224,7 @@ impl ExtensionManager {
                     );
                 }
                 let client = reqwest::Client::builder()
-                    .default_headers(default_headers)
+                    .default_headers(default_headers.clone())
```

Comment:
why not always send the bearer token if can acquire one along here? if we can acquire it, we need it anyway